graphite/graphite.go

```diff
@@ -26,12 +26,30 @@ type Series struct {
 
 type DataPoint []json.Number
 
+type Error struct {
+	subject string
+	body    string
+	err     error
+}
+
+func (e Error) Error() string {
+	return fmt.Sprintf("graphite: %s: %s", e.subject, e.err.Error())
+}
+
+func (e Error) Body() string {
```

Comment:
i like the idea of having private fields that can only be set at construction time, so we can guarantee they won't accidentally be modified and incorrect. Lemme know if you like this idea or still like to just export the fields directly.